### PR TITLE
fix(medusa): default sales channel for store variant endpoints

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- add middleware to include sales channel if not present in query params when querying variant endpoints

### CODE REVIEW

File: packages/medusa/src/api/routes/store/variants/index.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ export default (app) => {
 
   route.get(
     "/",
+    withDefaultSalesChannel({}),
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/store/variants/index.ts]: todo: make the argument optional

**Review by adrien2p — APPROVED** [review]
LGT one todo

**Comment by olivermrbl** [comment]
@pKorsholm can I get you to address the PR comment and make tests pass? 💪

**Comment by olivermrbl** [comment]
@pKorsholm have you tested this? It throws an error at the DB level

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**